File: 3420-find-occurrences-of-an-element-in-an-array/3420-find-occurrences-of-an-element-in-an-array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        List<Integer> ll = new ArrayList();
        for(int i=0;i<nums.length;i++){
            if(nums[i]==x){
                ll.add(i);
            }
        }
        int[] res  = new int[queries.length];

        for(int i=0;i<queries.length;i++){
            int occurenceIndex = queries[i]-1;
            if(occurenceIndex < ll.size()){
                res[i]=ll.get(occurenceIndex);
            }else{
                res[i]=-1;
            }
        }
        return res;
    }
}

File: 3420-find-occurrences-of-an-element-in-an-array/3420-find-occurrences-of-an-element-in-an-array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
    vector<int> arr;
    for(int i=0;i<nums.size();i++)
    {
        if(nums[i]==x)
        arr.push_back(i);
    }
    vector<int> res;
    for(int i=0;i<queries.size();i++)
    {
        int k=queries[i];
        if(k<=arr.size())
        {
            res.push_back(arr[k-1]);
        }
        else
        {
            res.push_back(-1);
        }
    }
    return res;
    }
};